pkg/api/v1/orchestrator/routes/statusValueKV.go

```diff
@@ -0,0 +1,93 @@
+package routes
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/metal-toolbox/conditionorc/internal/status"
+	"github.com/metal-toolbox/rivets/events/registry"
+	"github.com/nats-io/nats.go"
+	"github.com/pkg/errors"
+
+	rctypes "github.com/metal-toolbox/rivets/condition"
+)
+
+type statusValueKV interface {
+	publish(
+		facilityCode string,
+		conditionID uuid.UUID,
+		controllerID registry.ControllerID,
+		conditionKind rctypes.Kind,
+		newSV *rctypes.StatusValue,
+		create,
+		onlyTimestamp bool,
+	) error
+}
+
+// statusValue implements the statusValueKV interface
+type statusValue struct{}
+
+func initStatusValueKV() statusValueKV {
+	return &statusValue{}
+}
+
+func (s *statusValue) publish(
+	facilityCode string,
+	conditionID uuid.UUID,
+	controllerID registry.ControllerID,
+	conditionKind rctypes.Kind,
+	newSV *rctypes.StatusValue,
+	create,
+	onlyTimestamp bool,
+) error {
+	statusKV, err := status.GetConditionKV(conditionKind)
+	if err != nil {
+		return errors.Wrap(errPublishStatus, err.Error())
+	}
+
+	key := rctypes.StatusValueKVKey(facilityCode, conditionID.String())
+	// create
+	if create {
+		newSV.CreatedAt = time.Now()
+		if _, errCreate := statusKV.Create(key, newSV.MustBytes()); errCreate != nil {
+			return errors.Wrap(errPublishStatus, errCreate.Error())
+		}
+
+		return nil
+	}
+
+	// update
+	currEntry, err := statusKV.Get(key)
+	if err != nil && !errors.Is(err, nats.ErrKeyNotFound) {
+		return errors.Wrap(errPublishStatus, err.Error())
+	}
+
+	// only timestamp update takes the current value
+	var publishSV *rctypes.StatusValue
+	if onlyTimestamp {
+		curSV := &rctypes.StatusValue{}
+		if errJSON := json.Unmarshal(currEntry.Value(), &curSV); errJSON != nil {
+			return errors.Wrap(errUnmarshalKey, errJSON.Error())
+		}
+
+		if curSV.WorkerID != controllerID.String() {
+			return errors.Wrap(errControllerMismatch, curSV.WorkerID)
+		}
+
+		publishSV = curSV
+	} else {
+		if newSV == nil {
```

Comment:
This was implemented with tests as its own method so it could potentially be reused and exported for the rivet/events/controllers
Although, that final dedup was not done to have lesser diamond shaped dependencies. 
I'd like to leave this check in there, if we figure a way to reuse this method with rivets.
https://github.com/metal-toolbox/rivets/blob/d264c62cc3a8cf2c2ba7e24f912cc0c8a18c672d/events/controller/status.go#L102